### ISSUE TITLE
False positive when using function call for parameter default

### ISSUE THREAD

**Issue by dalen** [rodjek/puppet-lint#132]
When using a function call on the $name variable as the default for another parameter in a defined type you get a false positive for the optional parameters before required parameters check. For example in this code:

``` puppet
define foo($bar=extlookup($name)) {
}
```

This is using the dust_bunny branch
